### PR TITLE
Case 20535: renderLayer, primitiveMode, groupCulled Entity properties

### PR DESCRIPTION
**Pull request by SamGondelman**
https://highfidelity.manuscript.com/f/cases/20535/Add-renderLayer-primitiveMode-groupCulled-to-Entities-to-replace-Overlay-s-drawInFront-drawHUDLayer-isSolid-isGroupCulled

Note: primitiveMode only works on the entities that already support wireframe mode from luci.

Test plan:
- run the following tests:
- https://github.com/highfidelity/hifi_tests/tree/585564501e184c73c334d7964110a60e3d778d21/tests/engine/render/lod/entity/model
- https://github.com/highfidelity/hifi_tests/tree/585564501e184c73c334d7964110a60e3d778d21/tests/content/entity/shape/renderLayer
- https://github.com/highfidelity/hifi_tests/tree/585564501e184c73c334d7964110a60e3d778d21/tests/content/entity/model/renderLayer

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/16311/HighFidelity-Beta-PR14667-3341d75.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/16322/HighFidelity-Beta-PR14667-10068cc.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/16322/HighFidelity-Beta-Interface-PR14667-10068cc.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/16322/HighFidelity-Beta-PR14667-10068cc.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/16322/HighFidelity-Beta-Interface-PR14667-10068cc.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/16323/HighFidelity-Beta-PR14667-5dd59dd.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/16323/HighFidelity-Beta-PR14667-5dd59dd.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/16323/HighFidelity-Beta-Interface-PR14667-5dd59dd.dmg

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/16324/HighFidelity-Beta-PR14667-292071f.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/16324/HighFidelity-Beta-PR14667-292071f.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/16324/HighFidelity-Beta-Interface-PR14667-292071f.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/16324/HighFidelity-Beta-PR14667-292071f.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/16324/HighFidelity-Beta-Interface-PR14667-292071f.exe

File: libraries/networking/src/udt/PacketHeaders.h [hunk 1]
```diff
@@ -255,7 +255,12 @@ enum class EntityVersion : PacketVersion {
     MorePropertiesCleanup,
     FixPropertiesFromCleanup,
     UpdatedPolyLines,
-    FixProtocolVersionBumpMismatch
+    FixProtocolVersionBumpMismatch,
+    MigrateOverlayRenderProperties,
+
+    // Add new versions above here
+    NUM_PACKET_TYPE,
+    LAST_PACKET_TYPE = NUM_PACKET_TYPE - 1
```
Comment (AndrewMeadows) [inline libraries/networking/src/udt/PacketHeaders.h]: :+1:

File: libraries/shared/src/PrimitiveMode.cpp [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+};
+
+static const size_t PRIMITIVE_MODE_NAMES = (sizeof(primitiveModeNames) / sizeof(primitiveModeNames[0]));
+
+QString PrimitiveModeHelpers::getNameForPrimitiveMode(PrimitiveMode mode) {
+    if (((int)mode <= 0) || ((int)mode >= (int)PRIMITIVE_MODE_NAMES)) {
+        mode = (PrimitiveMode)0;
+    }
+
+    return primitiveModeNames[(int)mode];
```
Comment (AndrewMeadows) [inline libraries/shared/src/PrimitiveMode.cpp]: I see you are not on the `int32_t` bandwagon.

Comment (SamGondelman) [reply libraries/shared/src/PrimitiveMode.cpp]: This is just following a pattern established in our other property enum files, but I can change it latter everywhere at once.

**Comment by ZappoMan** [comment]
@shanzzam @samcake @SamGondelman @huffman - what's the plan for these properties and the "Create tools" - presumably these properties could be used in domain entities to create a particular visual effect. This PR does not appear to add these properties to the create tools property panel... was that intentional?

**Comment by SamGondelman** [comment]
@ZappoMan As I add these properties I'm putting up tickets for the Create team to expose them.  I still need to do that for the most recent batch of properties.

**Comment by SamGondelman** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/16690/HighFidelity-Beta-PR14667-292071f.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/16690/HighFidelity-Beta-PR14667-292071f.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/16690/HighFidelity-Beta-Interface-PR14667-292071f.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/16690/HighFidelity-Beta-PR14667-292071f.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/16690/HighFidelity-Beta-Interface-PR14667-292071f.exe

**Comment by qlimnavarro** [comment]
Currently Testing

**Comment by qlimnavarro** [comment]
Pass 
Auto Tests pass

**Pull request merged**